### ISSUE TITLE
Data Portability in Solid

### ISSUE THREAD

**Issue by melvincarvalho** [solid/solid#230]
There's been some chatter about the phrase on, solid [how it works](https://solid.inrupt.com/how-it-works)

> "Since you own your data, you’re free to move it at any time, without interruption of service."

There has also been a long standing issue on a similar topic at : https://github.com/solid/solid/issues/11

I think it's worth noting in that different people are going to have different views on this.  So as a history we used to be two projects, one called cross cloud (which had a message of data portability) and one called solid (which had a message of social linked data).  But now it's just solid.

Some folks are really attracted to the the portability use case, and others are attracted to the social use case.  

I think we still have two camps in solid.  Some think there are innovative solutions to portability, as a primary feature.  Some think that stability of URIs are actually a feature, and that portability is hard.

So I'm in the 2nd camp.  And so far, those that have tried to achieve portability have not got too far, imho.  I'm skeptical it can be done, but would love to see it happen.  But ultimately my experience of watching people try is that it's too hard a problem for the returns.

In my view the easiest way is to use a CNAME to redirect your own domain to a pod.  (This is also part of the reason I've suggested [.solid](http://solidns.org) domains).  But instead you can go for the "[Cool URIs dont change](https://www.w3.org/Provider/Style/URI) approach".

> _It is the the duty of a Webmaster to allocate URIs which you will be able to stand by in 2 years, in 20 years, in 200 years. This needs thought, and organization, and commitment_

I particularly am inspired by Tim's quote above.  Instead of portability choose a your URI (especially for your identity, but also for your spaces) well.  Discussion lately has made things clearer in my mind, in that this is a goal worth achieving, and a philosophy I'd like to try to help apply to solid.community.

Can portability be achieved?  I dont know.  Can stable URIs be achieved?  I dont know.  But hopefully we can have groups working on the different approaches, trying things, and see how far we get.

I think it's a good discussion, but bear in mind there's two schools of thought, and so worth exploring possibilities as politely as possible.

**Comment by kjetilk** [comment]
Yeah, I really think we need a kind of a virtual host feature. I myself have only got my webid on kjernsmo.net, because that is a domain that I control. We should probably also offer something like Apache's `.htaccess`, so that people can put in redirects at least.
